pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -451,12 +451,12 @@ func (r *ReconcileAWSFederatedAccountAccess) createOrUpdateIAMRole(awsClient aws
 				}
 
 				return role, nil
-
-			} else {
-				// Handle unexpected AWS API errors
-				return nil, errors.New(aerr.Code())
 			}
+			// Handle unexpected AWS API errors
+			return nil, errors.New(aerr.Code())
```

Comment:
One comment, in other places we explicitly log that this is an unknown error from AWS could you do the same? https://github.com/openshift/aws-account-operator/blob/master/pkg/controller/account/iam.go#L422

Reply:
Updated.  Adds a switch/case for known, and default for unknown, using the `util.LogAwsError` as requested.